### DISCUSSION TITLE
Unable to publish apps in Microsoft Store

### DISCUSSION THREAD

**Discussion by WenHong-Lau** [unoplatform/uno#17863] [Q&A]
Hi I am trying to publish my application to the Microsoft Store, it did not pass the validation as we are using the newer and unsupported version of Microsoft.Windows.SDK.BuildTools  (10.0.26100.1)

This is the error 
Package acceptance validation error: The package uses an unsupported version of file makepri.exe (10.0.26100.1). The following versions are allowed: < 10.0.10500.0; >= 10.0.10586.0 & < 10.0.11000.0; >= 10.0.14383.0 & < 10.0.14800.0; >= 10.0.15053.0 & < 10.0.15100.0; >= 10.0.16299.0 & < 10.0.16350.0; >= 10.0.17134.0 & < 10.0.17500.0; >= 10.0.17763.0 & < 10.0.18199.0; >= 10.0.18362.0 & < 10.0.18799.0; >= 10.0.19041.0 & < 10.0.19399.0; >= 10.0.20348.0 & < 10.0.21242.0; >= 10.0.22000.194 & < 10.0.22350.0; >= 10.0.22621.0 & < 10.0.25054.0. Please update your Visual Studio build tools and try again.

**Comment by MartinZikmund** [comment]
This issue is caused by https://github.com/microsoft/WindowsAppSDK/issues/4480. The workaround is to downgrade to older version of BuildTools for now

**Reply by WenHong-Lau** [reply]
How do I downgrade to older version of BuildTools? I don't see the option to downgrade it as it is a dependency of Uno SDK.

**Reply by morning4coffe-dev** [reply]
**@WenHong-Lau** You can follow this [docs page](https://platform.uno/docs/articles/features/using-the-uno-sdk.html?tabs=3rd-party-packages) to learn how to specify the desired version with `WinAppSdkBuildToolsVersion`. Alternatively, you can directly reference the package with the desired version in the `Directory.Packages.props` file, which should also work.

**Reply by WenHong-Lau** [reply]
I have added this line 10.0.19041.0 to the Directory.Build,props and it gave a lot errors. For exmaple,

"Detected package downgrade: Microsoft.Windows.SDK.BuildTools from 10.0.22621.756 to 10.0.19041.1. Reference the package directly from the project to select a different version.
UnoProj -> Uno.Extensions.Navigation.WinUI 4.1.24 -> Microsoft.Windows.SDK.BuildTools (>= 10.0.22621.756)
UnoProj -> Microsoft.Windows.SDK.BuildTools (>= 10.0.19041) UnoProj (net8.0-android), UnoProj (net8.0-android), UnoProj (net8.0-android), UnoProj (net8.0-android)"
